contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3807,12 +3836,32 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 				else if (is_alter_role_stmt(grant_role))
 				{
 					check_alter_role_stmt(grant_role);
+					GetUserIdAndSecContext(&save_userid, &save_sec_context);
 
+					if (!grant_role->is_grant && !grant_role->grantor)
+					{
+						/*
+						 * First execute REVOKE statement using bootstrap user.
+						 * This is needed since grantor of GRANTs from previous versions might
+						 * not be bbf_role_admin.
+						 */
+						SetUserIdAndSecContext(BOOTSTRAP_SUPERUSERID, save_sec_context | SECURITY_LOCAL_USERID_CHANGE);
+						PG_TRY();
+						{
+							call_prev_ProcessUtility(pstmt, queryString, readOnlyTree, context, params,
+													queryEnv, dest, qc);
+						}
+						PG_FINALLY();
+						{
+							/* Clean up. Restore previous state. */
+							SetUserIdAndSecContext(save_userid, save_sec_context);
+						}
+						PG_END_TRY();
```

Comment:
Can we identify those faulty GRANT and correct those in upgrade?

Reply:
Or Is it possible to escalate to BOOTSTRAP_SUPERUSERID in some of the cases only? Are there any alternatives apart from we escalate to BOOTSTRAP_SUPERUSERID -- using somewhat less-privileged user?

Reply:
Good suggestion, I too think it's better to fix the grantor in upgrade script instead of escalating to BOOTSTRAP user here. Updated the logic in newer commit.